x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/ranges/ranges.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiForm } from '@elastic/eui';
+
+import { autoInterval } from '../../../../../../../../src/plugins/data/common';
+import { RangeEditor } from './range_editor';
+import { OperationDefinition } from '../index';
+import { FieldBasedIndexPatternColumn } from '../column_types';
+import { updateColumnParam, changeColumn } from '../../../state_helpers';
+
+export const DEFAULT_INTERVAL = 1000;
+// TODO: find a way to read the UI settings config from here
+export const HISTOGRAM_MAX_BARS = 100;
+
+export const MODES = {
+  Range: 'range',
+  Histogram: 'histogram',
+} as const;
+
+export type MODES_TYPES = typeof MODES.Range | typeof MODES.Histogram;
+
+export interface RangeType {
+  from: number;
+  to: number;
+  label: string;
+}
+
+export interface RangeIndexPatternColumn extends FieldBasedIndexPatternColumn {
+  operationType: 'range';
+  params: {
+    type: MODES_TYPES;
+    interval: 'auto' | number;
+    maxBars: 'auto' | number;
+    intervalBase?: number;
+    ranges: RangeType[];
```

Comment:
If you want to reuse the types from esaggs, you could. See `src/plugins/data/common/search/aggs/buckets/histogram.ts` for example.